database/migrations/sql/20180603_create_initial_schema.sql

```diff
@@ -0,0 +1,461 @@
+);
+CREATE TRIGGER update_organisation_modified_at BEFORE UPDATE ON organisation
+  FOR EACH ROW EXECUTE PROCEDURE update_modified_at_column();
+
+
+CREATE TABLE organisation_subordinate (
+  parent_id BIGINT NOT NULL,
+  child_id  BIGINT NOT NULL,
+
+  CONSTRAINT organisation_subordinates_to_parent_organisation_fk FOREIGN KEY (parent_id) REFERENCES organisation ON DELETE CASCADE,
+  CONSTRAINT organisation_subordinates_to_child_organisation_fk  FOREIGN KEY (child_id)  REFERENCES organisation ON DELETE CASCADE
+);
+
+
+CREATE TABLE subscription (
+  subscription_id      SERIAL NOT NULL UNIQUE,
+  owner_id             BIGINT NOT NULL,
+  beneficiary_id       BIGINT NOT NULL,
+  frontline_account_id BIGINT NOT NULL,
+  subscription_type_id BIGINT NOT NULL,
+  expires_at           TIMESTAMP WITH TIME ZONE NOT NULL,
+  created_at           TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(),
+  modified_at          TIMESTAMP WITH TIME ZONE,
+  deleted_at           TIMESTAMP WITH TIME ZONE,
+
+  CONSTRAINT subscription_pk                             PRIMARY KEY (subscription_id),
+  CONSTRAINT subscription_to_owner_organisation_fk       FOREIGN KEY (owner_id)             REFERENCES organisation ON DELETE CASCADE,
+  CONSTRAINT subscription_to_beneficiary_organisation_fk FOREIGN KEY (beneficiary_id)       REFERENCES organisation,
+  CONSTRAINT subscription_to_frontline_account_fk        FOREIGN KEY (frontline_account_id) REFERENCES frontline_account,
+  CONSTRAINT subscription_to_subscription_type_fk        FOREIGN KEY (subscription_type_id) REFERENCES subscription_type
+);
+CREATE TRIGGER update_subscription_modified_at BEFORE UPDATE ON subscription
+  FOR EACH ROW EXECUTE PROCEDURE update_modified_at_column();
+
+
+CREATE TABLE training_session (
+  -- stuff
+);
+
+
+/*
+ * USER RELATED TABLES
+ */
+CREATE TABLE user_account (
+  user_account_id                  SERIAL NOT NULL UNIQUE,
+  user_name                        VARCHAR(100) NOT NULL,
+  user_secret                      VARCHAR(64) NOT NULL,
+  gender                           genders NOT NULL DEFAULT 'prefer not to say',
+  email                            VARCHAR(100),
+  phone_number                     VARCHAR(20),
+  post_code                        VARCHAR(10),
+  birth_year                       INT NOT NULL,
+  is_email_contact_consent_granted BOOLEAN NOT NULL DEFAULT false,
+  is_sms_contact_consent_granted   BOOLEAN NOT NULL DEFAULT false,
+  created_at                       TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(),
+  modified_at                      TIMESTAMP WITH TIME ZONE,
+  deleted_at                       TIMESTAMP WITH TIME ZONE,
+
+  CONSTRAINT user_account_pk PRIMARY KEY (user_account_id)
+);
+CREATE TRIGGER update_user_account_modified_at BEFORE UPDATE ON user_account
+  FOR EACH ROW EXECUTE PROCEDURE update_modified_at_column();
+
+
+CREATE TABLE access_role (
+  access_role_id   SERIAL NOT NULL UNIQUE,
+  access_role_name VARCHAR NOT NULL UNIQUE,
```

Comment:
Missing character limit, also we expecting the roles to change? Would this be better served with an enum?

Reply:
For fields that are not filled by user-supplied input, I think we can leave that character limit off. Means we wont constrain ourselves in future with arbitrary size limits that are a pain to change once in production.